api/src/services/attachment-service.ts

```diff
@@ -828,4 +848,126 @@ export class AttachmentService extends DBService {
 
     return { ...attachmentResult, key };
   }
+
+  /**
+   * Handle deletion of Project Attachment.
+   * If report:
+   * - delete authors
+   * - delete publish record
+   * - delete attachment
+   * If attachment:
+   * - delete publish record
+   * - delete attachment
+   * If attachment was published:
+   * - delete from Platform.
```

Comment:
Sorry, one last nit: The bottom 2 sections both say "If attachment" which makes it seem like the last point (about deleting from the platform) doesn't apply to reports, when it does.

I can help come up with other language if you want, but otherwise all I can recommend is to try and write this more verbosely (comments are free, so no need to be concise) and try and write them from the perspective of someone that doesn't know anything about SIMS attachments or the fact that we have 2 main types (Reports and Non-Reports).


The two kinds of attachments and even the relationship between deleting from SIMS and BioHub are business concepts, rather than technical concepts, so being really comment heavy is always worth it.